// IDL for New User Registration(for 2.0 architecture)

package com.caredear;

import "CommonUserCenter.proto";


message RegisterRequest{
    required RegLoginType    reg_type = 1;  // reg via mobile/email/name/etc..
    required DeviceType reg_device = 2; // which device?
    required string     reg_source = 3; // user reg via which App?
    required string     reg_name = 4;

    optional string     reg_password = 5;  // for name+password reg scenario
    optional string     reg_devbrand = 6;  // devices' brand(avaible for Android)
    optional int32      bypass_activation = 7; // set 1 to bypass activation step
}

// FIXME - result code be consistent with Web defination?
message RegisterResponse{
    required int32  result_code = 1; //CDS_XXX error code
    optional string extra_msg = 2; // further error msg if it has
    optional string reg_verifycode = 3; // for mobile, or email verification(if bypass_activation==0)
    optional string caredear_id = 4; // an unique ID for this new registered user.
}
